### PR TITLE
[G2M] add forwardRef wrapper to every component

### LINKED ISSUE #8: Add ForwardRef into every component
Allows the use of ref={} for DOM manipulation since the use case for with-ref.js was specifically created for tab-panels.

### PR DESCRIPTION
**Pull request by kc-leung**
fixed #8 

- Added forwardRef to all components
- Added displayName to all components for ref

### CODE REVIEW

File: src/alert.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
-import React from 'react'
+import React, { forwardRef } from 'react'
 import PropTypes from 'prop-types'
 import MUIAlert from '@material-ui/lab/Alert'
 import AlertTitle from '@material-ui/lab/AlertTitle'
 
-const Alert = ({ severity, message, header, width, ...props }) => {
+const Alert = forwardRef(({ severity, message, header, width, ...props }, ref) => {
```
Comment (parkdoeui) [inline src/alert.js:6]: why dont we add forwardRef at the end instead for a cleaner look?
```
export default forwardRef(Alert);
```

Comment (kc-leung) [reply src/alert.js:6]: Proptype warning

Screen Shot 2021-04-29 at 6 45 54 PM

Comment (DoParkEQ) [reply src/alert.js:6]: that's interesting 🤔

**Comment by woozyking** [comment]
can https://github.com/EQWorks/lumen-ui/blob/main/src/with-ref.js be reused for this purpose?

```js
import withRef from './with-ref'

const Alert = ...

export default withRef(Alert)
```

**Comment by kc-leung** [comment]
> can https://github.com/EQWorks/lumen-ui/blob/main/src/with-ref.js be reused for this purpose?
> 
> ```js
> import withRef from './with-ref'
> 
> const Alert = ...
> 
> export default withRef(Alert)
> ```

It is used in index.js and forwardRef is passed as a prop. In the way you suggest, the component will not have access to the ref if I am not wrong. If we use the with-ref then all components will probably have to be defined the same way as tab-panels which for me having forwardRef inside of each component looks cleaner

Screen Shot 2021-04-30 at 10 42 00 AM

**Comment by woozyking** [comment]
i see, fair enough

File: src/chip.js [hunk 2]
```diff
@@ -51,8 +51,8 @@ const useStyles = makeStyles((theme) => ({
     color: (styleProps) => `${styleProps.custom.backgroundColor} !important`,
   },
 }))
-
-const Chip = ({ clickable, color, custom, onDelete, rectangle, variant, margin, ...props }) => {
+forwardRef
```
Comment (woozyking) [inline src/chip.js]: typo?

Comment (kc-leung) [reply src/chip.js]: yes copy and paste typo

**Review by parkdoeui — APPROVED** [review]

**Review by woozyking — APPROVED** [review]

**Pull request merged**